packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/extension.tsp

```diff
@@ -0,0 +1,173 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Azure.ResourceManager.Models.Resources;
+
+model ExtensionsResource is ExtensionResource<ExtensionsResourceProperties> {
+  ...ResourceNameParameter<ExtensionsResource>;
+}
+
+/** ExtensionsResource properties */
+model ExtensionsResourceProperties {
+  @doc("The description of the resource.")
+  description?: string;
+
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@armResourceOperations
+interface ExtensionsResources {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/topLevelTrackedResources/top/providers/Azure.ResourceManager.Models.Resources/extensionsResources/extension
```

Comment:
is it possible that we could construct a path like
```
/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/extensionsResources/extension
```
via the extension resource feature?

Reply:
as far as I know, the `ExtensionResource` model is like a "scope" resource which generates the resource like:
```
/{resourceUri}/providers/{namespace}/foos/{name}
```
and since its resourceUri could be any valid existing resource, it could be a subscription or resource group, we should add some verification for these scenario as well

Reply:
With existing feature, we could, at least for Java. Though it's not a valid extension resource.

Reply:
> it could be a subscription or resource group, we should add some verification for these scenario as well

Yes, you are right! We could add case for tenant, subscription, resource group and normal resources.

Reply:
@v-hongli1 Consider adding scenarios for another two scopes: subscription and resource groups.

You probably need:

1. Refer below https://github.com/Azure/cadl-ranch/blob/c5f897d13996f5de2a078ddf8f127570bb59fead/packages/cadl-ranch-specs/http/azure/client-generator-core/access/mockapi.ts#L35-L54

for adding more than 1 mockapis to one scenario.

2. In extension.tsp, update each `@scenarioDoc` to indicate that three expected scopes are needed to pass this scenario test.

Reply:
Fixed by commit https://github.com/Azure/cadl-ranch/commit/8ca02c564ae744600ca1d7dbfba100606086d852